### DISCUSSION TITLE
Reply and copy buttons on hover

### DISCUSSION THREAD

**Discussion by testsnake** [Chatterino/chatterino2#5490] [Ideas]
Twitch has a reply button when hovering over a message, and 7tv adds to this with a button that copies the given message.

![image](https://github.com/Chatterino/chatterino2/assets/12944119/d12fe8fa-ad2d-433a-8361-624b406a66db)

While Chatterino does have a reply button, it's always on and is quite distracting. 

![image](https://github.com/Chatterino/chatterino2/assets/12944119/213575a8-d1f7-474a-98c0-7c57530b3ae9)

Adding this hover functionality to Chatterino would make it significantly more usable for me, and probably at least 1 other person.

![image](https://github.com/Chatterino/chatterino2/assets/12944119/c27b4892-f4c4-48e2-be7f-f85cbd4a93a7)

**Comment by Mm2PL** [comment]
Pretty sure the reply button is ugly like this is because of technical issues. As for copying I don't think there was a proposal for that before. We just use menu options for that.

**Comment by Felanbird** [comment]
As outlined in the PR that added replies, an easy way for us to add an on hover button was not found, that's why we don't have one.

We only added the always-on reply button so users _could_ have it if they wanted it, we are aware it's ugly that's why it's disabled by default.

**Reply by Felanbird** [reply]
https://github.com/Chatterino/chatterino2/pull/3722

**Comment by Nerixyz** [comment]
This does work, and I shouldn't be doing this, but whatever:

https://github.com/Chatterino/chatterino2/assets/19953266/0d53f4e4-6734-4672-aa0c-4031701d530a

https://github.com/Nerixyz/chatterino2/commit/238f62f2504872e68831f7544b5808de691254f8 - this is based on a button refactoring I want to eventually get in. And in the future, this will use ✨ SvgButton ✨ which can adapt to the theme (by providing two sources).

**Reply by dnsge** [reply]
Moving a widget around is kinda cursed but this actually looks pretty nice. Hoverable buttons would add a lot of possibilities for new features and ways to interact with chatterino. For example, holding shift or control while hovering could show a different set of buttons that do more advanced things.

**Reply by testsnake** [reply]
Even if cursed, I'd love to see this in a future release!
